### PR TITLE
feat (api): update secret backend mechanism, remove hard coded vault

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

File: engine/api/repositoriesmanager/repositories_manager.go [hunk 1]
```diff
@@ -28,23 +28,15 @@ var (
...
-			log.Warning("RepositoriesManager> Unable to get a Vault client %s\n", err)
-			return err
-		}
-
-		secrets, err := vaultClient.GetSecrets()
-
-		if err != nil {
-			log.Warning("RepositoriesManager> Unable to fetch Vault secrets %s\n", err)
-			return sdk.ErrSecretKeyFetchFailed
+		secrets := secretClient.GetSecrets()
```
Comment (proullon) [inline engine/api/repositoriesmanager/repositories_manager.go]: why not double return value ?

File: engine/api/secret/secretbackend/types.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	gob.Register(Options{})
+	gob.Register(Secrets{})
+	gob.Register(SecretError(""))
+}
+
+//Driver is a plugin interface for retrieve CDS secrets
+type Driver interface {
+	Init(MapVar) error
+	Name() string
+	GetSecrets() Secrets
```
Comment (proullon) [inline engine/api/secret/secretbackend/types.go]: Why not "GetSecrets() (Secrets, error)

File: engine/api/repositoriesmanager/repositories_manager.go (hunk 1 above)
Comment (fsamin) [inline engine/api/repositoriesmanager/repositories_manager.go]: go-plugin seems to be quite unstable with double return values
one of this days i should go through protobuf and grpc...

File: engine/api/secret/secretbackend/types.go (hunk 2 above)
Comment (fsamin) [inline engine/api/secret/secretbackend/types.go]: go-plugin seems to be quite unstable with double return values
one of this days i should go through protobuf and grpc...

**Comment by sguiheux** [comment]
LGTM

**Pull request merged**